test/distributed/fsdp/test_fsdp_misc.py

```diff
@@ -118,6 +118,35 @@ def test_device_id_auto_wrap(self):
                 torch.device("cuda", torch.cuda.current_device()),
             )
 
+    @skip_if_lt_x_gpu(2)
+    def test_fsdp_device_id_cpu_offload(self):
+        """
+        Ensures that even if device_id is specified but we have
+        CPU offload, module is on CPU after init.
+        """
+        class MyModel(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = nn.Linear(10, 10)
+                self.b = nn.Linear(10, 10)
+
+            def forward(self, x):
+                return self.b(self.a(x))
+
+        model = MyModel()
+
+        fsdp = FSDP(
+            model,
+            auto_wrap_policy=always_wrap_policy,
+            cpu_offload=CPUOffload(offload_params=True),
+            device_id=torch.cuda.current_device()
+        )
+
+        cpu_device = torch.device("cpu")
+        for fsdp_unit in FSDP.fsdp_modules(fsdp):
+            fsdp_param = next(fsdp_unit.parameters())
```

Comment:
nit: This is subtly relies on the order of `.parameters()`. Maybe we should do `fsdp_param = fsdp_unit.params[0]`?

Reply:
Curious, what does `.params` give if the current FSDP unit manages no parameters?

Reply:
it is just an empty list then. It is not safe to access `.params[0]` in general, but I recommended it since it was fine for your particular model.